src/cci/cci_handle_mng.c

```diff
@@ -1205,7 +1205,7 @@ hm_create_health_check_th (char useSSL)
   rv = pthread_attr_setdetachstate (&thread_attr, PTHREAD_CREATE_DETACHED);
   rv = pthread_attr_setscope (&thread_attr, PTHREAD_SCOPE_SYSTEM);
 #endif /* WINDOWS */
-  rv = pthread_create (&health_check_th, &thread_attr, hm_thread_health_checker, (void *) useSSL);
+  rv = pthread_create (&health_check_th, &thread_attr, hm_thread_health_checker, (void *) (size_t) useSSL);
```

Comment:
hm_thread_health_checker handling of argument may be modified for clarity (at cci_handle_mng.c:1476):
```c
  char useSSL = *((char *) (&arg));
```
to:
```c
  char useSSL = ((size_t) arg) != 0 ? USESSL : NON_USESSL;
```